### ISSUE TITLE
deal with inconsistent use of .rda extension

### ISSUE THREAD

**Issue by cjyetman** [2DegreesInvesting/PACTA_analysis#124]
In some cases, RDS files are saved with the `.rda` extension. This causes a bunch of confusion, and breaks functionality of,  for instance, `read_in_file()`

https://github.com/2DegreesInvesting/PACTA_analysis/blob/acdedabadccdcdeb05cc2e0384f9c78e77a5abd6/0_global_functions.R#L24-L43

**Comment by cjyetman** [comment]
for background (copy-pasted from a Slack discussion)...

`.rda` is a formerly used extension for a R workspace image file, which can contain 1 or more R objects, and objects stored in that file can be added to your current environment using base R functions load or attach, and they are typically created by using base R function `save.image()` or `save()`

`.RData` is the default extension currently used for the same exact files as `.rda` files

`.rds` is the extension for RDS files created by base R function `saveRDS()` and read by base R function `readRDS()` . These are importantly different from `.rda`/`.RData` files in the following ways…
- they are saved and read by totally different base R functions
- they can only save one, single R object (versus `.RData` which can save many objects)
- when reading in an `.rds` file, the value of the object can be saved as any user assigned object name, this is very much different than loading an `.RData` file which will load into your environment all of the objects it contains and overwrite any existing objects you have in your environment if the saved names of the objects conflict. This is one (imo) really nice advantage to using (proper) `.rds` files because you can do something like `new_data <- readRDS("mydata.rds")`  rather than having to do `load("mydata.RData")` and have whatever objects saved in there overwrite whatever they conflict with in your current environment.

At 2dii, unfortunately we have this long history of saving RDS files with an `.rda` extension. Now it seems we’re doing both…. saving RDS files with `.rda`, and also saving RData files with `.rda`.

that is why this works
`load("data/currencies.rda")`
but this does not
`load("data-raw/bench_regions.rda")`

and this works
`readRDS("data-raw/bench_regions.rda")`
but this does not
`readRDS("data/currencies.rda")`

**Comment by cjyetman** [comment]
and this is very likely the issue causing problems with `read_file_in()` as stated in #119

**Comment by maurolepore** [comment]
@cjyetman, I closed #119 because I think it's a duplicate. I'll follow up here, starting with this summary of #119:

* I recently moved the problematic .rda files from data/ to data-raw/, then saved them back into data/ via the recommended approach: `usethis::use_data()` (somewhat explained at https://github.com/2DegreesInvesting/PACTA_analysis/issues/119#issuecomment-704218835). 

* Indeed, this brakes `read_in_file()` which currently treats .rda as .rds, and does several other things that are currently unused, which makes the function larger than needed and prone to errors. I suggest we replace it or prune it (https://github.com/2DegreesInvesting/PACTA_analysis/issues/119#issuecomment-704232497). 

* @jacobvjk said:

> ... I don't seem to understand all the implications of using `devtools::load_all()`. 

`devtools::load_all()` is a package-aware wrapper around `load()`, so the implication to consider is the one that @cjyetman highlighted above: `load()` directly assigns a name to an environment. If we prefer control assignment explicitely with `<-` we should wrap it in a function that implements something like this:

``` r
  
e <- rlang::new_environment()
load("data/bench_regions.rda", envir = e)
result <- e$bench_regions

head(result)
#>   scenario_geography        country country_iso reg_count
#> 1             Global    Afghanistan          AF       281
#> 2             Global        Albania          AL       281
#> 3             Global        Algeria          DZ       281
#> 4             Global American Samoa          AS       281
#> 5             Global        Andorra          AD       281
#> 6             Global         Angola          AO       281
```

<sup>Created on 2020-10-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

This is a much simpler (less error prone) implementation than the current `read_in_file()`.
